### ISSUE TITLE
table column can not be editted

### ISSUE THREAD

**Issue by vonschbo** [olifolkerd/tabulator#42]
Hello, you mentioned in
Editable Data
Columns of the table can be set as editable using the editable parameter in the column definition. (see Define Column Headers for more details).

When I write this to add new column, editable does not work.
    thiscolumns.push({
          title: "" + newCol,
          field: "" + newCol,
          fitColumns: true,
          sorter: "string",
          editable: true
        });

**Comment by olifolkerd** [comment]
To add a new column you need to use the ***addColumn*** function, if you just push it onto the column definition array directly then Tabulator does not know to re-render the table.
```js
$("#example-table").tabulator("addColumn", {title:"Age", field:"age"}, true, "name");
```
See the [Add Column Documentation](http://olifolkerd.github.io/tabulator/docs/#movable) for full details on this function.

**Comment by vonschbo** [comment]
I am currently using a database to store these inputs, I only find the way to store it is through push it into an array objects.

**Comment by olifolkerd** [comment]
Any changes to column definition data have to be passed in through a function otherwise they wont trigger the re render functions that correctly bind the editors etc to the cells.

If you want you can pass the entire array of column definitions into Tabulator after your push, using the ***setColumns*** function

```js
var thiscolumns = [
    {title:"Name", field:"name", sortable:true, sorter:"string", width:200, editable:true},
    {title:"Age", field:"age", sortable:true, sorter:"number", align:"right", formatter:"progress"},
    {title:"Height", field:"height", sortable:true, formatter:"star", align:"center", width:100},
    {title:"Favourite Color", field:"col", sorter:"string", sortable:false},
    {title:"Date Of Birth", field:"dob", sortable:true, sorter:"date", align:"center"},
],

$("#example-table").tabulator("setColumns", thiscolumns, false) //completely overwrite existing columns with new columns definition array
```

Full details on this can be found in [the documentation](http://olifolkerd.github.io/tabulator/docs/#movable)

**Comment by vonschbo** [comment]
I mean edit the column header.

**Comment by olifolkerd** [comment]
Ahhh, now i follow you.

The column header text is not editable, it is a fixed value.

Are you looking to be able to edit the column header then get the new value, or just to let users tweak their column headers.

**Comment by vonschbo** [comment]
I am looking to be able to edit the column header then get the new value.

**Comment by olifolkerd** [comment]
This is not possible through a column definition directly, but it should be achievable though a bit of jiggery pokery with the value of column title.

I will put together a demo and send it over this evening

**Comment by olifolkerd** [comment]
i have just released version 2.4 which now includes the features you are after.

You need to set the ***editableTitle*** parameter to ***true*** in the definition object for each column you wish to be able to edit. This will add an input element around the title so users can edit it.

When a column title is edited the ***colTitleChanged*** callback is triggered.

```js
("#example-table").tabulator({

    columns:[
        {title:"Name", field:"name", editableTitle:true},
        {title:"Age", field:"age", editableTitle:true},
        {title:"Height", field:"height", editableTitle:true},
        {title:"Favourite Color", field:"col", editableTitle:true},
        {title:"Date Of Birth", field:"dob", editableTitle:true},
    ],
    colTitleChanged:function(value, col, cols){
        //value - the new column value
        //col - the updated column definition object for the affected column
        //cols - the updated column definition array for all columns in the table
    },
});
```

For more Information, check out the [Editable Column Title Documentation](http://olifolkerd.github.io/tabulator/docs/#editable-column-title).
